extensions/jsonb/deployment/src/main/java/io/quarkus/jsonb/deployment/JsonbProcessor.java

```diff
@@ -90,6 +90,10 @@ void build(BuildProducer<ReflectiveClassBuildItem> reflectiveClass,
         for (AnnotationInstance deserializeInstance : index.getAnnotations(JSONB_TYPE_DESERIALIZER)) {
             registerInstance(reflectiveClass, deserializeInstance);
         }
+
+        // register String constructors for reflection as they may not have been properly registered by default
+        // see https://github.com/quarkusio/quarkus/issues/10873
```

Comment:
Can we add the corresponding link to the Yasson issue too so that it is easier to quickly check it ?

Reply:
There is no corresponding fix on Yasson.
In fact, we don't know why it didn't works so it's difficult to provides a fix on Yasson because Optionnal seems to be correctly handled there.

Reply:
Yes, I understand that this is not fixed yet in Yasson. What I meant is to add the "corresponding upstream issue" in the comment.

Reply:
Sorry, I said fix instead of issue.
There is no corresponding issue upstream.

Reply:
Ah no worries. 

I was hoping that there is an upstream issue for better support of `Optional<String>`. It makes tracking easier but I suppose we can live with what we have.

Reply:
> There is no corresponding issue upstream.

Well, shouldn't we create one and add a link pointing to it?

Reply:
Already merge this one.
I'll copy your comment to the original issue so we can continue working on the root cause inside it.